lib/hexpm/web/controllers/dashboard_controller.ex

```diff
@@ -28,6 +28,50 @@ defmodule Hexpm.Web.DashboardController do
     end
   end
 
+  def keys(conn, _params) do
+    user = conn.assigns.current_user
+    keys = Keys.all(user)
+
+    render_keys(conn, keys)
+  end
+
+  def revoke_key(conn, %{"name" => name} = params) do
+    user = conn.assigns.current_user
+    key = Hexpm.Repo.one(Key.get(user, params["name"]))
+
+    if key do
+      case Hexpm.Repo.update(Key.revoke(key)) do
+        {:ok, _struct} ->
+          conn
+          |> put_flash(:info, "The key #{params["name"]} was revoked successfully")
+          |> redirect(to: Routes.dashboard_path(conn, :keys))
+        {:error, _changeset} ->
```

Comment:
please mirror how we handle validations error e.g. when adding new email address in http://localhost:4000/dashboard/email: on error, we re-render the form and add errors inline. See errors are handled when adding empty email address.